src/App.js

```diff
@@ -1,20 +1,441 @@
+            keyTakeaways:
+              keyTakeaways !== undefined ? keyTakeaways : task.keyTakeaways,
+          };
+        }
+        return task;
+      });
+      return { tasksCompleted: reflectedTasks };
+    }, this.updateLocalStorage);
+  };
+
+  deleteTaskCompleted = (id) => {
+    this.setState(
+      (prevState) => ({
+        tasksCompleted: prevState.tasksCompleted.filter(
+          (task) => task.id !== id
+        ),
+      }),
+      this.updateLocalStorage
+    );
+  };
+
+  // Move task from In Review to Completed
+  moveTaskInReview = (id) => {
+    const taskToMove = this.state.tasksInReview.find((task) => task.id === id);
+    if (taskToMove) {
+      const newId = this.state.tasksCompleted.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+      this.setState(
+        (prevState) => ({
+          tasksInReview: prevState.tasksInReview.filter(
+            (task) => task.id !== id
+          ),
+          tasksCompleted: [...prevState.tasksCompleted, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-        </header>
+      <div className="app-container">
+        <Header className="header" />
+        <div className="task-container">
+          <TaskList
+            tasks={this.state.tasks}
+            updateTask={this.updateTask}
+            addTaskToDo={this.addTaskToDo}
+            deleteTask={this.deleteTask}
+            resetLocalStorage={this.resetLocalStorage}
+            moveTaskOpen={this.moveTaskOpen}
+          />
+          <TaskListInProgress
+            tasksInProgress={this.state.tasksInProgress}
+            updateTask={this.updateTaskInProgress}
+            addTaskToDo={this.addTaskToDoInProgress}
+            deleteTask={this.deleteTaskInProgress}
+            resetLocalStorage={this.resetLocalStorageInProgress}
+            moveTaskInProgress={this.moveTaskInProgress}
+            moveTaskInProgressBacktoOpen={this.moveTaskInProgressBacktoOpen}
+          />
+          <TaskListInReview
+            tasksInReview={this.state.tasksInReview}
+            updateTask={this.updateTaskInReview}
+            addTaskToDo={this.addTaskToDoInReview}
+            deleteTask={this.deleteTaskInReview}
+            moveTaskInReview={this.moveTaskInReview}
+            resetLocalStorage={this.resetLocalStorageInReview}
+            moveTaskInReviewBackToInProgress={
+              this.moveTaskInReviewBackToInProgress
+            }
+          />
+          <TaskListCompleted
+            tasksCompleted={this.state.tasksCompleted}
+            deleteTask={this.deleteTaskCompleted}
+            reflectionToDo={this.reflectionToDo}
+            resetLocalStorage={this.resetLocalStorageCompleted}
```

Comment:
This is somewhat painful to read with lots of repetition. We definitely would be able to refactor this. The first thing that comes to mind here is the not define a function for each status separately. updating and deleting, should be not tied to the status. You should rather use status parameters to decide the status, but only have a single function for update delete and creation. Same goes for local storage reset or moving tasks.